dev/preview/previewctl/pkg/preview/service.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package preview
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+var errSvcNotReady = errors.New("proxy service not ready")
+
+const proxySvcName = "proxy"
+
+func (p *Preview) getVMProxySvcStatus(ctx context.Context) error {
+	_, err := p.harvesterKubeClient.CoreV1().Namespaces().Get(ctx, p.namespace, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(errSvcNotReady, err.Error())
+	}
+
+	svc, err := p.harvesterKubeClient.CoreV1().Services(p.namespace).Get(ctx, proxySvcName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(errSvcNotReady, err.Error())
+	}
```

Comment:
> Hey Aleks, really brilliant work here ✨! I wasn't expecting that we would get deeper into Kubernetes go-client intricacies so soon with `previewctl`, but you found a very good use for it with the watch control-loop 🙂. Not only that but you also introduced kubevirt's go-client as well
> 
> I believe that the introduction of KubeVirt's lib and k8s informers is great for us. We'll need them for future projects and we can get used to them while `previewctl` is still simple 😉 With that said, this PR is quite big and my lack of experience with KubeVirt's lib and informers will block me from giving you a super detailed review 😅. I really want to get better at it tho, future reviews will be better! Maybe @liam-j-bennett's experience with k8s controllers could help us here :)
> 
> For now, I focused on understanding the changes made to the `Preview` struct, why they were needed and made some comments about it.
> 
> I understand you had to make changes to some commands to make the kubernetes clients work as you expected, but I think the abstractions look a bit weird 🤔.
> 
> Would it make sense to split the Kubernetes clients and Preview into separate objects? Something like:
> 
> ```go
> type Harvester struct {
> 	kClient kubernetes.Interface
> 	dynamicClient dynamic.Interface
>         previews []Preview
> 
> 	logger *logrus.Entry
> }
> 
> type Preview struct {
> 	branch                 string
> 	name                   string
> 	namespace              string
>         vm                     kubervirtv1.VirtualMachine // Not sure about this one... Maybe we could reuse their library for SSH access? https://github.com/kubevirt/kubevirt/blob/f60c13703d4bc73fe4ebb3c44c6d8379d6ba0577/pkg/virtctl/ssh/ssh.go
> }
> ```
> 
> I really don't want to sound like a blocker here, just trying to surface some ideas to get the abstraction in a good shape so they can be easily enhanced in future interactions :)

Thanks for the feedback! I agree that it would make more sense to have the k8s components split out - I'll do this in a following PR. I also want to change some things around the informers, as currently it's not strictly correct the way they are being used - but I did it this way as it was quicker and it kind kind of grew out a bit more than I intended. Also didn't see the `virtctl/ssh` package 👍  I think that will make our lives easier. I'll see how it works and add it with the work of porting the `install-context` :)